pom.xml

```diff
@@ -3220,9 +3217,34 @@
 
     <profile>
       <id>scala-2.12</id>
+      <activation>
+        <activeByDefault>true</activeByDefault>
+      </activation>
       <build>
         <pluginManagement>
           <plugins>
+            <plugin>
+              <groupId>net.alchim31.maven</groupId>
+              <artifactId>scala-maven-plugin</artifactId>
+              <version>4.4.0</version>
+              <configuration>
+                <args combine.children="append">
+                    <arg>-unchecked</arg>
+                    <arg>-Xfatal-warnings</arg>
+                    <arg>-deprecation</arg>
+                    <arg>-Ywarn-unused-import</arg>
+                    <arg>-P:silencer:globalFilters=.*deprecated.*</arg>
```

Comment:
Do we need to silence deprecation here? that seems separate and a larger step

Reply:
Oh, actually I think it's matched with the Scala side actually: https://github.com/apache/spark/blob/master/project/SparkBuild.scala#L221-L226
So I guess it might be fine because this is to make Maven work in the same way.

Reply:
Ok fine if it's primarily about consistency